File: Loom.Core/Models/ModelMetaData.vb
Namespace Models
    ''' <summary>
    ''' Select the best model for a provider
    ''' </summary>
    Public Class ModelMetaData
        Public Property ModelId As String
        Public Property ProviderName As String
        Public Property ContextWindow As Integer
        Public Property SupportsTools As Boolean
        Public Property CostLevel As Integer ' 1 (cheap) -> 5 (expensive)
        Public Property Capabilities As List(Of String) ' ('vision', 'audio'...)
    End Class
End Namespace

File: implementations/csharp/ModelMetaData.cs
using System.Collections.Generic;

namespace Models
{
    /// <summary>
    /// Select the best model for a provider
    /// </summary>
    public class ModelMetaData
    {
        public string ModelId { get; set; }
        public string ProviderName { get; set; }
        public string ContextWindow {  get; set; }
        public bool SupportsTools { get; set; }
        public int CostLevel { get; set; } // 1 (cheap) -> 5 (expensive)
        public List<string> Capabilities { get; set; } // (vision, audio, ...)
    }
}
